config/src/main/java/com/typesafe/config/ConfigFactory.java

```diff
@@ -1090,6 +1092,89 @@ public static Config parseResourcesAnySyntax(String resourceBasename) {
         return parseResourcesAnySyntax(resourceBasename, ConfigParseOptions.defaults());
     }
 
+    /**
+     * Parse only any application replacement (specified by one of config.{resource,file,url}), returning
+     * an empty Config if no overrides were set.
+     *
+     * @since 1.4.1
+     *
+     * @return a {@link java.util.Optional} containing any specified replacement, or {@link Optional#empty()}
+     * if none was specified.
+     */
+    public static java.util.Optional<Config> parseApplicationReplacement() {
+        return parseApplicationReplacement(ConfigParseOptions.defaults());
+    }
+
+    /**
+     * Like {@link #parseApplicationReplacement()} but allows you to specify a class loader
+     * ti yse rather than the current context class loader.
+     *
+     * @since 1.4.1
+     *
+     * @param loader the class loader
+     * @return a {@link java.util.Optional} containing any specified replacement, or {@link Optional#empty()}
+     * if none was specified.
+     */
+    public static java.util.Optional<Config> parseApplicationReplacement(ClassLoader loader) {
+        return parseApplicationReplacement(ConfigParseOptions.defaults().setClassLoader(loader));
+    }
+
+    /**
+     * Like {@link #parseApplicationReplacement()} but allows you to specify parse options.
+     *
+     * @since 1.4.1
+     *
+     * @param parseOptions parse options
+     * @return a {@link java.util.Optional} containing any specified replacement, or {@link Optional#empty()}
+     * if none was specified.
+     */
+    public static java.util.Optional<Config> parseApplicationReplacement(ConfigParseOptions parseOptions) {
+        ensureClassLoader(parseOptions, "parseApplicationReplacement");
+        ClassLoader loader = parseOptions.getClassLoader();
```

Comment:
Oh, oops - I didn't catch this before merging, but ensureClassLoader returns the new options, doesn't modify the original options.

Reply:
Oops. I can open a follow up fix PR today.

Reply:
https://github.com/lightbend/config/pull/709